src/js/components/Calendar/Calendar.js

```diff
@@ -268,47 +257,94 @@ const Calendar = forwardRef(
           }
         });
         return dateArray;
       }
       return undefined;
     };
 
+    let timestamp = timestampProp;
+    if (timestampProp === undefined) {
+      if (Array.isArray(dateProp) && dateProp.length) {
+        if (Array.isArray(dateProp[0])) {
+          timestamp = getTimestamp(dateProp[0][0]);
+        } else timestamp = getTimestamp(dateProp[0]);
+      } else if (typeof dateProp === 'string') {
+        timestamp = getTimestamp(dateProp);
+      } else if (Array.isArray(datesProp) && datesProp.length) {
+        if (Array.isArray(datesProp[0])) {
+          timestamp = getTimestamp(datesProp[0][0]);
+        } else timestamp = getTimestamp(datesProp[0]);
+      } else if (typeof datesProp === 'string') {
+        timestamp = getTimestamp(datesProp);
+      } else if (typeof referenceProp === 'string') {
+        timestamp = getTimestamp(referenceProp);
+      }
+    }
+
+    const normalizedDate = useMemo(
+      () =>
+        timestampProp === undefined
+          ? normalizeDate(dateProp, timestamp)
+          : dateProp,
+      [dateProp, timestamp, timestampProp],
+    );
+
+    const normalizedDates = useMemo(
+      () =>
+        timestampProp === undefined
+          ? normalizeDate(datesProp, timestamp)
+          : datesProp,
+      [datesProp, timestamp, timestampProp],
+    );
+
     // set date when caller changes it, allows us to change it internally too
-    const [date, setDate] = useState(dateProp);
+    const [date, setDate] = useState(normalizedDate);
     useEffect(() => {
-      setDate(normalizeDate(dateProp));
-    }, [dateProp]);
+      setDate(normalizedDate);
+    }, [normalizedDate]);
 
     // set dates when caller changes it, allows us to change it internally too
-    const [dates, setDates] = useState(datesProp);
+    const [dates, setDates] = useState(normalizedDates);
     useEffect(() => {
-      setDates(normalizeDate(datesProp));
-    }, [datesProp]);
+      setDates(normalizedDates);
+    }, [normalizedDates]);
 
     // set reference based on what the caller passed or date/dates.
     const [reference, setReference] = useState(
-      normalizeReference(referenceProp, date, dates),
+      normalizeReference(
+        referenceProp,
+        normalizedDate,
+        normalizedDates,
+        timestamp,
+      ),
     );
     useEffect(
       () =>
-        setReference(normalizeReference(referenceProp, dateProp, datesProp)),
-      [dateProp, datesProp, referenceProp],
+        setReference(
+          normalizeReference(
+            referenceProp,
+            normalizedDate,
+            normalizedDates,
+            timestamp,
+          ),
+        ),
+      [referenceProp, normalizedDate, normalizedDates, timestamp],
     );
 
     // normalize bounds
     const [bounds, setBounds] = useState(
-      boundsProp ? boundsProp.map((b) => normalizeForTimezone(b)) : undefined,
+      boundsProp ? boundsProp.map((b) => b) : undefined,
```

Comment:
Can we just `useState(boundsProp)`? It doesn't look like the `map()` is doing anything.